File: testdata/samplecdp/config/artichoke/submissions/accepted/artJB.cpp
#include <iostream>
#include <cmath>
#include <cstdio>
using namespace std;

int main()
{
	long long p, a, b, c, d;
	int n;

	cin >> p;
	cin >> a >> b >> c >> d;
	cin >> n;
	double smallest = sin(a*n+b) + cos(c*n+d);
	double maxdec = 0.0;
	int iend, istart, ismall = n;
	for(int i=n-1; i>=1; i--) {
		double val = sin(a*i+b) + cos(c*i+d);
		if (val - smallest >= maxdec) {
			maxdec = val - smallest;
			istart = i;
			iend = ismall;
		}
		else if (val < smallest) {
			smallest = val;
			ismall = i;
		}
	}
//	if (maxdec == 0.0)
//		cout << 0 << endl;
//	else
//		cout << p*maxdec << ' ' << istart << ' ' << iend << endl;
//		cout << p*maxdec << endl;
		printf("%.6f\n", p*maxdec);
}

File: testdata/samplecdp/config/artichoke/submissions/accepted/artJB.java
import java.util.Scanner;

public class artJB
{
    public static void main(String [] args)
    {
        long p, a, b, c, d;
        int n;
        Scanner in = new Scanner(System.in);

        p = in.nextLong();
        a = in.nextLong();
        b = in.nextLong();
        c = in.nextLong();
        d = in.nextLong();
        n = in.nextInt();
        double smallest = Math.sin(a*n+b) + Math.cos(c*n+d);
        double maxdec = 0.0;
        int iend=0, istart=0, ismall = n;
        for(int i=n-1; i>=1; i--) {
            double val = Math.sin(a*i+b) + Math.cos(c*i+d);
            if (val - smallest >= maxdec) {
                maxdec = val - smallest;
                istart = i;
                iend = ismall;
            }
            else if (val < smallest) {
                smallest = val;
                ismall = i;
            }
        }
//        if (maxdec == 0.0)
//            System.out.println("0 0 0");
//        else
//            System.out.println(p*maxdec + " " + istart + " " + iend);
            System.out.println(p*maxdec);
    }
}

